WorkAndFun/src/main/resources/log4j2.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="WARN">
+    <Properties>
+        <Property name="LOG_EXCEPTION_CONVERSION_WORD">%xwEx</Property>
+        <Property name="LOG_LEVEL_PATTERN">%3p</Property>
+        <Property name="LOG_DATEFORMAT_PATTERN">yyyy-MM-dd HH:mm:ss.SSS</Property>
+        <Property name="CONSOLE_LOG_PATTERN">%clr{%d{${sys:LOG_DATEFORMAT_PATTERN}}}{faint} %clr{${sys:LOG_LEVEL_PATTERN}} %clr{%pid}{magenta} %clr{---}{faint} %clr{[%15.15t]}{faint} %clr{%-40.40c{1.}}{cyan} %clr{:}{faint} %m%n${sys:LOG_EXCEPTION_CONVERSION_WORD}</Property>
+        <Property name="FILE_LOG_PATTERN">%d{${LOG_DATEFORMAT_PATTERN}} ${LOG_LEVEL_PATTERN} %pid --- [%t] %-40.40c{1.} : %m%n${sys:LOG_EXCEPTION_CONVERSION_WORD}</Property>
+        <Property name="CONSOLE_LOG_CHARSET">UTF-8</Property>
+        <Property name="FILE_LOG_CHARSET">UTF-8</Property>
+        <Property name="LOG_PATH">logs</Property>
+        <Property name="LOG_FILE">log4j2-app.log</Property>
+        <Property name="EXERCISE_FILE_LOG_PATTERN">%d{HH:mm:ss.SSS} ${LOG_LEVEL_PATTERN} : %m%n${sys:LOG_EXCEPTION_CONVERSION_WORD}</Property>
+    </Properties>
+
+    <Appenders>
+        <Console name="Console" target="SYSTEM_OUT" follow="true">
+            <PatternLayout pattern="${sys:CONSOLE_LOG_PATTERN}" charset="${sys:CONSOLE_LOG_CHARSET}"/>
+        </Console>
+
+        <RollingFile name="LogFile" fileName="${sys:LOG_PATH}/${sys:LOG_FILE}"
+                     filePattern="${sys:LOG_PATH}/$${date:yyyy-MM}/app-%d{yyyy-MM-dd-HH}-%i.log.gz">
+            <PatternLayout pattern="${sys:FILE_LOG_PATTERN}" charset="${sys:FILE_LOG_CHARSET}"/>
+            <Policies>
+                <SizeBasedTriggeringPolicy size="10 MB"/>
+            </Policies>
+        </RollingFile>
+    </Appenders>
+
+    <Loggers>
+        <Logger name="pl.infoshare.workandfun.mainpage.Index" level="info"/>
+        <Logger name="pl.infoshare.workandfun.announcements.AnnouncementController" level="info"/>
+        <Logger name="pl.infoshare.workandfun.announcements.AnnouncementService" level="info"/>
+        <Logger name="pl.infoshare.workandfun.exception.AnnouncementNotFoundException" level="info"/>
+        <Logger name="pl.infoshare.workandfun.announcements.AnnouncementService" level="debug"/>
+        <Logger name="pl.infoshare.workandfun.announcements.AnnouncementAPI" level="debug"/>
+        <Logger name="pl.infoshare.workandfun.announcements.mappers.AddAndEditMapper" level="trace"/>
+        <Logger name="pl.infoshare.workandfun.announcements.mappers.QuickViewAnnouncementMapper" level="trace"/>
+
+        <Root level="info">
+            <AppenderRef ref="Console" level="info"/>
+            <AppenderRef ref="LogFile" level="trace"/>
```

Comment:
moim zdaniem linia 41, na konsolę warto wyrzucać od levelu debug - tak myślę

a linia 42 do log file warto logować od lvlu info. - w praktyce jak jest na trace to zaraz bedzie w logfile straszny śmietnik i niczego nie znajdziesz, a do tego logfile będzie strasznie szybko puchł, przy dużych apkach jak logi mają po kilkaset MB albo nawet kilka GB nie widziałem w nich nawet logów info, wręcz tylko od error w górę - ale na potrzeby kursu myślę że można pchać od info.
trace jest tak nieistotny ,że się całkowicie pomija

Reply:
Z trace w logfile miałem podobne przemyślenia, jak zobaczyłem, że odpalenie strony głównej pobiera wszystkie ogłoszenia z bazy i zmienia w dto, co generuje 3 wpisy trace logów - ładnie pliczek by puchł.